controllers/djob_controller_databricks.go

```diff
@@ -121,6 +123,90 @@ func (r *DjobReconciler) refresh(instance *databricksv1alpha1.Djob) error {
 	return r.Update(context.Background(), instance)
 }
 
+
+/*
+IsDJobUpdated  is a method to check if the cluster has the latest version of a certain Djob
+*/
```

Comment:
It's a great idea to compare existing Djob with new Djob however, with this approach of comparing we need to make sure to update the method if we add/update Djob .
what was the result of  `reflect.DeepEqual(instance.Spec, jobExisting.Spec)` 
I was looking into https://github.com/kubernetes-sigs/kubebuilder/issues/592 and seems checking Spec should work

Reply:
@Azadehkhojandi  At the very beginning, I tried this approach to compare the Spec for both new djob and existing djob, however, for some reason it doesn't work always returns a wrong result, I dug into it, I found the json package doesn't provide a way to access nested values for nested objects, so I ended up with my approach to manually compare them.

I tried it again yesterday, here is the result:
	oldComparsion := reflect.DeepEqual(ExistingDJob, newDJob)
	r.Log.Info(fmt.Sprintf("oldComparsion %v", oldComparsion))
	newComparsion := reflect.DeepEqual(instance.Spec, jobExisting.Settings)
	r.Log.Info(fmt.Sprintf("newComparsion %v", newComparsion))
![image](https://user-images.githubusercontent.com/16556831/72765718-fb026380-3c41-11ea-8be2-4263074d88a4.png)

Reply:
The other solution here as it mentioned in the link you shared "[kubernetes-sigs/kubebuilder#592](https://github.com/kubernetes-sigs/kubebuilder/issues/592)" is to Hash the entire Spec and store it somewhere e.g. Kubernetes volume or something else for the first time and the second time gets the stored Spec compared with the new Spec

Reply:
What about hashing the Spec and storing it in an annotation on the djob?

Reply:
@stuartleeks yeah I went to this solution, however, I am still working on it, for some reason, I was able to store the hashing, but I can't retrieve it, still in progress.

Reply:
@Azadehkhojandi Done a while ago forgot to mention.
Hashing the spec and storing in annotations.